extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +184,123 @@ private void restartJobs() {
         if (schedulerFuture != null && !schedulerFuture.isCancelled()) {
             schedulerFuture.cancel(true);
         }
-
-        schedulerFuture = scheduler.schedule(new Runnable() {
-            @Override
-            public void run() {
+        schedulerFuture = scheduler.schedule(() -> {
+            monitor.lock();
+            try {
                 stopJobs();
-
-                try {
-                    synchronized (schedulerLock) {
-                        if (quartzScheduler == null) {
-                            quartzScheduler = StdSchedulerFactory.getDefaultScheduler();
+                if (getThing().getStatus() == ONLINE) {
+                    String thingUID = getThing().getUID().toString();
+                    String typeId = getThing().getThingTypeUID().getId();
+                    // Daily Job
+                    Job dailyJob = getDailyJob();
+                    if (dailyJob == null) {
+                        logger.error("Daily Job instance not present");
+                        return;
+                    }
+                    if (scheduledExecutor == null) {
+                        logger.warn("Thread Pool Executor is not available");
+                        return;
+                    }
+                    Expression midNightExpression = new CronExpression(DAILY_MIDNIGHT);
+                    if (addJobToQueue(dailyJob)) {
+                        scheduledExecutor.schedule(dailyJob, midNightExpression);
+                    }
+                    logger.info("Scheduled astro job-daily-{} at midnight for thing {}", typeId, thingUID);
+
+                    // Startup Job
+                    LocalDateTime currentTime = LocalDateTime.now();
+                    Expression cronExpression = new CronExpression(createCronFromTemporal(currentTime));
```

Comment:
I agree that calling `run` directly could be a blocking the current thread. However, by looking at the content of `PositionalJob.run` and `DailyJob.run` I don't see a problem here.

If you decide to start 2 new `Thread`s everytime the `restartJobs` is called, then where do you remove them again? Otherwise we will get a lot of zombie threads, don't we?